msgraph-developer-proxy/ChaosEngine.cs

```diff
@@ -269,6 +288,12 @@ public class ChaosEngine {
                 UpdateProxyResponse(e, responseComponents, matchingResponse);
         }
 
+        private static void AddProxyHeader(Request r) {
+            if (r.Headers is not null) {
```

Comment:
Should we initialize the headers collection if it is null? 
For graph requests we know that will be a 401, so should we even bother to forward the request to graph?